providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java

```diff
@@ -0,0 +1,296 @@
+                evaluatedValue = toValue(dynamicConfig);
+                break;
+            case LAYER:
+                Layer layer = fetchLayer(user, featureConfig);
+                evaluatedValue = toValue(layer);
+                break;
+            default:
+                break;
+        }
+        return ProviderEvaluation.<Value>builder()
+            .value(evaluatedValue)
+            .build();
+    }
+
+    @SneakyThrows
+    protected DynamicConfig fetchDynamicConfig(StatsigUser user, FeatureConfig featureConfig) {
+        return Statsig.getConfigAsync(user, featureConfig.getName()).get();
+    }
+
+    @SneakyThrows
+    protected Layer fetchLayer(StatsigUser user, FeatureConfig featureConfig) {
+        return Statsig.getLayerAsync(user, featureConfig.getName()).get();
+    }
+
+    private Value toValue(DynamicConfig dynamicConfig) {
+        MutableContext mutableContext = new MutableContext();
+        mutableContext.add("name", dynamicConfig.getName());
+        mutableContext.add("value", Structure.mapToStructure(dynamicConfig.getValue()));
+        mutableContext.add("ruleID", dynamicConfig.getRuleID());
+        mutableContext.add("groupName", dynamicConfig.getGroupName());
+        List<Value> secondaryExposures = new ArrayList<>();
+        dynamicConfig.getSecondaryExposures().forEach(secondaryExposure -> {
+            Value value = Value.objectToValue(secondaryExposure);
+                secondaryExposures.add(value);
+            }
+        );
+        mutableContext.add("secondaryExposures", secondaryExposures);
+        return new Value(mutableContext);
+    }
+
+    private Value toValue(Layer layer) {
+        MutableContext mutableContext = new MutableContext();
+        mutableContext.add("name", layer.getName());
+        mutableContext.add("value", Structure.mapToStructure(layer.getValue()));
+        mutableContext.add("ruleID", layer.getRuleID());
+        mutableContext.add("groupName", layer.getGroupName());
+        List<Value> secondaryExposures = new ArrayList<>();
+        layer.getSecondaryExposures().forEach(secondaryExposure -> {
+                Value value = Value.objectToValue(secondaryExposure);
+                secondaryExposures.add(value);
+            }
+        );
+        mutableContext.add("secondaryExposures", secondaryExposures);
+        mutableContext.add("allocatedExperiment", layer.getAllocatedExperiment());
+        return new Value(mutableContext);
+    }
+
+    @NotNull
+    private static FeatureConfig parseFeatureConfig(EvaluationContext ctx) {
+        Value featureConfigValue = ctx.getValue(FEATURE_CONFIG_KEY);
+        if (featureConfigValue == null) {
+            throw new IllegalArgumentException("feature config not found at evaluation context.");
+        }
+        if (!featureConfigValue.isStructure()) {
+            throw new IllegalArgumentException("feature config is not a structure.");
+        }
```

Comment:
Probably need to document these exceptions in readme so users add the required entries.

Reply:
It is documented in readme:
> String/Integer/Double evaluations evaluation gets [Dynamic config](https://docs.statsig.com/server/javaSdk#reading-a-dynamic-config) or [Layer](https://docs.statsig.com/server/javaSdk#getting-an-layerexperiment) evaluation.
  As the key represents an inner attribute, feature config is required as a parameter with data needed for evaluation.
  For an example of dynamic config of product alias, need to differentiate between dynamic config or layer, and the dynamic config name.